Source/WebKit/WebProcess/Plugins/PDF/UnifiedPDF/UnifiedPDFPlugin.mm

```diff
@@ -3874,6 +3877,95 @@ static NSStringCompareOptions compareOptionsForFindOptions(WebCore::FindOptions
         wantsWheelEventsChanged();
 }
 
+#if PLATFORM(IOS_FAMILY)
+
+void UnifiedPDFPlugin::setSelectionRange(FloatPoint pointInRootView, TextGranularity granularity)
+{
+    RetainPtr selection = [&] -> PDFSelection * {
+        auto pointInPlugin = convertFromRootViewToPlugin(roundedIntPoint(pointInRootView));
+        auto pointInDocument = convertDown<FloatPoint>(CoordinateSpace::Plugin, CoordinateSpace::PDFDocumentLayout, pointInPlugin);
+        auto pageIndex = m_presentationController->nearestPageIndexForDocumentPoint(pointInDocument);
+        auto pointInPage = convertDown(CoordinateSpace::PDFDocumentLayout, CoordinateSpace::PDFPage, pointInDocument, pageIndex);
+        RetainPtr page = m_documentLayout.pageAtIndex(pageIndex);
+        switch (granularity) {
+        case TextGranularity::WordGranularity:
+            return [page selectionForWordAtPoint:pointInPage];
+        case TextGranularity::LineGranularity:
+            return [page selectionForLineAtPoint:pointInPage];
+        default:
+            // Unsupported selection granularities.
+            return nil;
+        }
+    }();
+
+    if (!selection)
+        return;
+
+    setCurrentSelection(WTFMove(selection));
```

Comment:
I think you want to unconditionally set the selection?

Reply:
So I didn't want to clear the selection if it was an unsupported granularity, but perhaps I can unconditionally set it if I use the other method above.

Reply:
oh right, I see what you mean, but then we don't do the right thing if `selectionForWordAtPoint:` returns `nil` (when we have word granularity).

Reply:
Maybe return `std::optional<PDFSelection *>` in the lambda? `std::nullopt` for unsupported granularities, which we can use for an early return

Reply:
Oh, I ended up falling back to a default granularity of `.character` (with a debug assertion), and then unconditionally call into `setCurrentSelection()` — I think that should make the logic a bit simpler to follow.